src/main/java/mathandel/backend/service/EditionService.java

```diff
@@ -92,33 +105,47 @@ public ApiResponse editEdition(EditionTO editionTO, Long editionId, Long userId)
         return new ApiResponse("Edition edited successfully");
     }
 
-    public List<EditionTO> getEditions() {
-        return mapEditions(editionRepository.findAll());
-    }
-
-    private List<EditionTO> mapEditions(List<Edition> all) {
-        return all.stream().map(e -> new EditionTO()
-                .setId(e.getId())
-                .setName(e.getName())
-                .setDescription(e.getDescription())
-                .setEndDate(e.getEndDate())
-                .setNumberOfParticipants(e.getParticipants().size())
-                .setMaxParticipants(e.getMaxParticipants())).collect(Collectors.toList());
+    public List<EditionTO> getEditions(Long userId) {
+        return mapEditions(editionRepository.findAll(), userId);
     }
 
     public ApiResponse makeUserEditionModerator(Long userId, Long editionId, String username) {
         User moderator = userRepository.findById(userId).orElseThrow(() -> new AppException("User does not exist"));
         Edition edition = editionRepository.findById(editionId).orElseThrow(() -> new ResourceNotFoundException("Edition", "id", editionId));
-        User requestedUser =  userRepository.findByUsername(username).orElseThrow(() -> new ResourceNotFoundException("User", "username", username));
+        User requestedUser = userRepository.findByUsername(username).orElseThrow(() -> new ResourceNotFoundException("User", "username", username));
 
-        if(!edition.getModerators().contains(moderator)) {
+        if (!edition.getModerators().contains(moderator)) {
             throw new BadRequestException("You have no access to this resource");
         }
-        if(!requestedUser.getRoles().contains(new Role().setName(RoleName.ROLE_MODERATOR))) {
+        if (!edition.getParticipants().contains(requestedUser)) {
```

Comment:
a musi byc?

Reply:
no nie musi ale wtedy powinienem go wrzucic jako participanta (panel edycji bedzie zawierał funkcjonalnosc participanta + ew funkcjonalnosc moderatora). A co jesli jest przekroczona maksymalna liczba uzytkownikow?? wtedy bedzie problem a jak juz jest uzytkownikiem to wszystko spoko